rows.go

```diff
@@ -229,30 +231,23 @@ func (scd *snowflakeChunkDownloader) Next() ([]*string, error) {
 		if scd.CurrentChunkIndex > 1 {
 			scd.Chunks[scd.CurrentChunkIndex-1] = nil // detach the previously used chunk
 		}
-		scd.ChunksMutex.Unlock()
-
-		ticker := time.NewTicker(time.Second)
-		for range ticker.C {
-			scd.ChunksMutex.Lock()
+		for scd.Chunks[scd.CurrentChunkIndex] == nil {
```

Comment:
This should be `for`, because the broadcast signal indicates other chunk's completion. For example, the main thread here waits for the chunk 1 to finish, but the broadcast signal could come from chunk 2 first, then this should be retried.